contracts/ERC721Universal.sol

```diff
@@ -13,69 +16,133 @@ import "./IERC721Universal.sol";
  *  The null address is the only address that cannot own any slot; as usual,
  *  it is used as the target address of the transfer executed within the burn method.
  */
-contract ERC721Universal is IERC721Universal, ERC721 {
+contract ERC721Universal is
+    IERC721Universal,
+    IERC721UpdatableBaseURI,
+    IERC721Broadcast,
+    ERC721,
+    Ownable
+{
+    /// @inheritdoc IERC721Universal
+    uint32 public constant ERC721UniversalVersion = 1;
+
+    /// @inheritdoc IERC721UpdatableBaseURI
+    bool public isBaseURILocked;
 
     // the map that returns true for tokens that have been burned
-    mapping(uint256 tokenId => bool) public isBurnedToken;
+    mapping(uint256 tokenId => bool) public isBurned;
 
-    // the string prepended to tokenId to return tokenURI
-    string public baseURI;
+    // this string is prepended to tokenId to form the tokenURI
+    string private _baseURIStorage;
 
     // the string used to insert tokenId when building a universal location
     string private constant TOKENID_STR = "GeneralKey(";
 
     constructor(
-        string memory name,
-        string memory symbol,
+        address owner_,
+        string memory name_,
+        string memory symbol_,
         string memory baseURI_
-    ) ERC721(name, symbol) {
-        baseURI = baseURI_;
+    ) ERC721(name_, symbol_) Ownable(owner_) {
+        _baseURIStorage = baseURI_;
         emit NewERC721Universal(address(this), baseURI_);
     }
 
+    /// @inheritdoc IERC721UpdatableBaseURI
+    function updateBaseURI(string calldata newBaseURI) external onlyOwner {
+        if (isBaseURILocked) revert BaseURIAlreadyLocked();
+        _baseURIStorage = newBaseURI;
+        emit UpdatedBaseURI(newBaseURI);
+    }
+
+    /// @inheritdoc IERC721UpdatableBaseURI
+    function lockBaseURI() external onlyOwner {
+        if (isBaseURILocked) revert BaseURIAlreadyLocked();
+        isBaseURILocked = true;
+        emit LockedBaseURI(_baseURIStorage);
+    }
+
+    /// @inheritdoc IERC721Broadcast
+    function broadcastMint(uint256 tokenId) external {
+        _broadcast(tokenId, address(0));
```

Comment:
why zero instead of current owner to whom was minted?

Reply:
you're confusing the parameter: the second input to _broadcast is the `from` address. The `to` needs not be specified, since it's forced to be the owner, as it couldn't be otherwise.